src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestMessage.cs

```diff
@@ -101,29 +99,16 @@ public HttpMethod Method
             get { return _requestUri; }
             set
             {
-                if ((value != null) && (value.IsAbsoluteUri) && (!HttpUtilities.IsHttpUri(value)))
-                {
-                    throw new ArgumentException(HttpUtilities.InvalidUriMessage, nameof(value));
-                }
                 CheckDisposed();
 
-                // It's OK to set 'null'. HttpClient will add the 'BaseAddress'. If there is no 'BaseAddress'
-                // sending this message will throw.
+                // It's OK to have a 'null' request Uri. If HttpClient is used, the 'BaseAddress' will be added.
+                // If there is no 'BaseAddress', sending this request message will throw.
+                // Note that we also allow the string to be empty: null and empty are considered equivalent.
```

Comment:
This comment seems more pertinent to the constructor overload that takes a string, below. Move it there?